src/AzureAppConfigurationImpl.ts

```diff
@@ -29,20 +37,49 @@ export class AzureAppConfigurationImpl extends Map<string, unknown> implements A
         super();
         // Enable request tracing if not opt-out
         this.requestTracingEnabled = requestTracingEnabled();
-        if (this.requestTracingEnabled) {
-            this.enableRequestTracing();
-        }
 
         if (options?.trimKeyPrefixes) {
             this.sortedTrimKeyPrefixes = [...options.trimKeyPrefixes].sort((a, b) => b.localeCompare(a));
         }
+
+        if (options?.refreshOptions?.enabled) {
+            const { watchedSettings, refreshIntervalInMs } = options.refreshOptions;
+            // validate watched settings
+            if (watchedSettings === undefined || watchedSettings.length === 0) {
+                throw new Error("Refresh is enabled but no watched settings are specified.");
+            }
+
+            // custom refresh interval
+            if (refreshIntervalInMs !== undefined) {
+                if (refreshIntervalInMs < MinimumRefreshIntervalInMs) {
+                    throw new Error(`The refresh interval time cannot be less than ${MinimumRefreshIntervalInMs} milliseconds.`);
+                } else {
+                    this.refreshInterval = refreshIntervalInMs;
+                }
+            }
+
+            this.sentinels = watchedSettings.map(setting => {
+                if (setting.key.includes("*") || setting.label?.includes("*")) {
+                    throw new Error("The character '*' is not supported in key or label.");

```

Comment:
Watch settings with wildcard key or label is not supported

Reply:
I was following python's impl checking wildcard char `*`. In .NET there's no such check for watched settings. Characters `*`  and `,` are just regarded as plain text, and thus won't match any valid setting in most of time.

So now I'm thinking whether we need the check.
- Option a) keep the check. Then we probably also want to add `,` as well.
- Option b) remove the check. Then we need to document or add doc comments to the API.

Reply:
Now that we are planning collection monitoring support, I think it's a good time to consider if we want to block these characters at all. Let's sync offline about this.

Reply:
The idea is `watchedSettings` takes a list of specific configuration setting ids. To watch wildcard patterns I think we should resort to collection monitoring. 
In this PR I will take option a) to block wildcard chars like `*` and `,`.  And in the error message I will highlight "watched settings" and specific chars, as suggested.

Reply:
updated in https://github.com/Azure/AppConfiguration-JavaScriptProvider/pull/21/commits/1da9647555fbbf19d049d10a02ed5f18a7ea513e